### PR TITLE
Implement a default protocol

### PR DESCRIPTION
**Pull request by morgangiraud**
Connect #175 

A first system to implement a default protocol, revealing its public function to the `tfe` module.

Tests are incoming. each one is already working, but since this changes introduce a global state to the module, they break when run in parallel. I need to find a way to make them self-contained.

### CODE REVIEW

File: examples/matmul.py [hunk 1]
```diff
@@ -22,15 +22,17 @@
 # Setting your custom configuration
 # tfe.set_config(config)
 
-with tfe.protocol.Pond() as prot:
+# The default protocol is called pond, it's a 3 party computation protocol
```
Comment (mortendahl) [inline examples/matmul.py]: grammatically these two lines could be cleaned up a bit :P

File: tensorflow_encrypted/__init__.py [hunk 2]
```diff
@@ -1,27 +1,72 @@
...
-from .protocol import global_caches_updator
+from .protocol import global_caches_updator, Pond, get_protocol
 from .player import player
 from . import io
 from . import protocol
 from . import layers
 from . import convert
 
 
+def get_protocol_public_func(prot: protocol.Protocol) -> list:
```
Comment (mortendahl) [inline tensorflow_encrypted/__init__.py]: is this called somewhere or pure black magic?

File: tensorflow_encrypted/protocol/__init__.py [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
-# from unencrypted_fixedpoint import UnencryptedFixedpoint
 from .pond import Pond, TFEVariable, TFETensor
 from .securenn import SecureNN
 
-from .protocol import Protocol, global_caches_updator, memoize
+from .protocol import Protocol, global_caches_updator, memoize, set_protocol, get_protocol
+
+
+def get_all_funcs() -> list:
+    all_prot_methods = []
```
[the remark this answers is not in the document]
Comment (morgangiraud) [reply tensorflow_encrypted/protocol/__init__.py]: Done for the list. 

Currently, I need to instantiate as some of the functions are not pure functions and rely on some states inside the classes.

**Comment by mortendahl** [comment]
@morgangiraud ready to merge this guy?

**Pull request merged**